aws/session/session.go

```diff
@@ -440,6 +440,15 @@ func mergeConfigSrcs(cfg, userCfg *aws.Config, envCfg envConfig, sharedCfg share
 			cfg.Credentials = credentials.NewStaticCredentialsFromCreds(
 				envCfg.Creds,
 			)
+
+		} else if len(envCfg.WebIdentityTokenFilePath) > 0 {
```

Comment:
Should this also validate that the RoleArn env var is provided, or just let the creds fail?

Reply:
The role ARN is a required field, I'd say we should fail without it.

Reply:
This will return an error, if role ARN is empty, when `Retrieve` is called. I figure it'd be best to add it to the credential chain and if it is empty return an error signaling an issue back to the user.

Reply:
I am okay with changing it to behaving the other way, if it seems more preferable.

Reply:
How about this, return an error if only one is set at the session create level rather than at the credential retrieval. That way it fails quickly and let's the user know that they have misconfigured the application

Reply:
Ah ok that makes sense. I like an error being reported if only one is present